File: backend/GameStudy/src/main/java/ru/nsu/GameStudy/models/Task.java
package ru.nsu.GameStudy.models;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Time;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name="title")
    private String title;
    @Column(name="description")
    private String description;
    @ManyToOne
    @JoinColumn(name="discipline_id", referencedColumnName = "id")
    private Discipline discipline;
    @Column(name="time")
    private Time time;
    @Column(name="experience")
    private Integer experience;
    @Column(name="complexity")
    private Integer complexity;
    @Column(name="answer")
    private String answer;
}

File: frontend/gamestudy/src/types/apps/task.ts
export interface TaskType {
    id?: number;
    title?: string;
    description?: string;
    time?: string;
    experience?: number;
    complexity?: number;
    answer?: string;
    discipline_id?: number;
    course_ids?: number[];
    lesson_ids?: number[];
    task_result_ids?: number[];
    game_method_id?: number;
    video_path?: string;
}